pkg/catalog/install.go

```diff
@@ -0,0 +1,103 @@
+package catalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+// Writer takes a profile and writes it out into a medium.
+type Writer interface {
+	Output(prof *profilesv1.ProfileSubscription) error
+}
+
+// FileWriter is a Writer using a file as backing medium.
+type FileWriter struct {
+	Filename string
+}
+
+// Output writes the profile subscription yaml data into a given file.
+func (fw *FileWriter) Output(prof *profilesv1.ProfileSubscription) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	f, err := os.OpenFile(fw.Filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer func(f *os.File) {
+		if err := f.Close(); err != nil {
+			fmt.Printf("Failed to close file %s\n", f.Name())
+		}
+	}(f)
+	if err := e.Encode(prof, f); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Install using the catalog at catalogURL and a profile matching the provided profileName generates a profile subscription
+// writing it out with the provided profile subscription writer.
+func Install(catalogURL, catalogName, profileName, subName, namespace, branch, configMap string, writer Writer) error {
```

Comment:
that is quite a lengthy signature, do we want a struct instead?

Reply:
Yes, definitely. I can do that.